src/m365/outlook/commands/message/message-remove.spec.ts

```diff
@@ -0,0 +1,249 @@
+    sinonUtil.restore([accessToken.isAppOnlyAccessToken]);
+    sinon.stub(accessToken, 'isAppOnlyAccessToken').returns(true);
+
+    await assert.rejects(command.action(logger, { options: { id: messageId } } as any),
+      new CommandError(`The option 'userId' or 'userName' is required when removing a message using application permissions`));
+  });
+
+  it('throws an error when both userId and userName are defined when removing a message using application permissions', async () => {
+    sinonUtil.restore([accessToken.isAppOnlyAccessToken]);
+    sinon.stub(accessToken, 'isAppOnlyAccessToken').returns(true);
+
+    await assert.rejects(command.action(logger, { options: { id: messageId, userId: userId, userName: userPrincipalName } } as any),
+      new CommandError(`Both options 'userId' and 'userName' cannot be set when removing a message using application permissions`));
+  });
+
+  it('throws an error when userId is not a valid GUID when removing a message using application permissions', async () => {
+    sinonUtil.restore([accessToken.isAppOnlyAccessToken]);
+    sinon.stub(accessToken, 'isAppOnlyAccessToken').returns(true);
+
+    await assert.rejects(command.action(logger, { options: { id: messageId, userId: 'foo' } } as any),
+      new CommandError(`The value 'foo' for 'userId' option is not a valid GUID`));
+  });
+
+  it('throws an error when userName is not a valid user principal name when removing a message using application permissions', async () => {
+    sinonUtil.restore([accessToken.isAppOnlyAccessToken]);
+    sinon.stub(accessToken, 'isAppOnlyAccessToken').returns(true);
+
+    await assert.rejects(command.action(logger, { options: { id: messageId, userName: 'foo' } } as any),
+      new CommandError(`The value 'foo' for 'userName' option is not a valid user principal name`));
+  });
+
+  it('throws an error when both userId and userName are defined when removing a message using delegated permissions', async () => {
+    await assert.rejects(command.action(logger, { options: { id: messageId, userId: userId, userName: userPrincipalName } } as any),
+      new CommandError(`Both options 'userId' and 'userName' cannot be set when removing a message using delegated permissions`));
+  });
+
+  it('throws an error when userId is not a valid GUID when removing a message using delegated permissions', async () => {
+    await assert.rejects(command.action(logger, { options: { id: messageId, userId: 'foo' } } as any),
+      new CommandError(`The value 'foo' for 'userId' option is not a valid GUID`));
+  });
+
+  it('throws an error when userName is not a valid user principal name when removing a message using delegated permissions', async () => {
+    await assert.rejects(command.action(logger, { options: { id: messageId, userName: 'foo' } } as any),
+      new CommandError(`The value 'foo' for 'userName' option is not a valid user principal name`));
+  });
+
+  it('throws an error when the message cannot be found', async () => {
```

Comment:
It also handles other errors, so let's rename this test to something like `correctly handles API errors`.